### PR TITLE
[INLONG-10720][Sort] Add Elasticsearch6 connector on Flink 1.18

### LINKED ISSUE #10720: [Feature][Sort] Add Elasticsearch6 connector on Flink 1.18
### Description

Add Elasticsearch6 connector on Flink 1.18 to sort-flink-v1.18
The parent issue is #10467 

### Use case

User can sink data to elasticsearch6 through this connector.

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by XiaoYou201**
### [INLONG-10720][Sort] Add Elasticsearch6 connector on Flink 1.18

Fixes #10720 

### Motivation
There is no es connector in sort-connectors， but es is a popular distributed, RESTful search and analytics engine.

So, this pr will add the es6 connector.

### Modifications

- Reference the official connector flink-connector-elasticsearch6 and it will support send audit information later.

### Documentation

  - Does this pull request introduce a new feature? (yes)

### CODE REVIEW

**Comment by XiaoYou201** [comment]
As follow image, I implement es 6 connector and  postgreSQL to ElasticSearch6 test successfully . But due to the format-base module not compatible with flink 1.18, So, I have to submit the connector code.The test result as follow.
image
image
image

File: inlong-sort/sort-flink/sort-flink-v1.18/sort-connectors/elasticsearch6/pom.xml [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        <dependency>
+            <groupId>org.apache.inlong</groupId>
+            <artifactId>sort-flink-dependencies-v1.18</artifactId>
+            <version>${project.version}</version>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.flink</groupId>
+            <artifactId>flink-connector-elasticsearch6</artifactId>
+            <version>3.0.1-1.17</version>
```
Comment (EMsnap) [inline inlong-sort/sort-flink/sort-flink-v1.18/sort-connectors/elasticsearch6/pom.xml:46]: please make a property

Comment (XiaoYou201) [reply inlong-sort/sort-flink/sort-flink-v1.18/sort-connectors/elasticsearch6/pom.xml]: fix it, thx~

**Review by EMsnap — APPROVED** [review]

**Comment by EMsnap** [comment]
Please fix the compile error on workflow in format-base for flink-1.18 and submit a pr later

**Review by aloyszhang — APPROVED** [review]

**Pull request merged**